src/databricks/labs/lsql/backends.py

```diff
@@ -146,15 +146,14 @@ def fetch(self, sql: str) -> Iterator[Row]:
         return self._sql.fetch_all(sql)
 
     def save_table(self, full_name: str, rows: Sequence[DataclassInstance], klass: Dataclass, mode="append"):
-        if mode == "overwrite":
-            msg = "Overwrite mode is not yet supported"
-            raise NotImplementedError(msg)
         rows = self._filter_none_rows(rows, klass)
         self.create_table(full_name, klass)
         if len(rows) == 0:
             return
         fields = dataclasses.fields(klass)
         field_names = [f.name for f in fields]
+        if mode == "overwrite":
+            self.execute(f"TRUNCATE TABLE {full_name}")
         for i in range(0, len(rows), self._max_records_per_batch):
```

Comment:
we'd need to properly support overwrites in the raw sql shape to keep the same'ish semantics as Spark:
1. `INSERT INTO {full_name}_tmp ...`
2. `CREATE OR REPLACE TABLE {full_name} AS SELECT * FROM {full_name}_tmp`
3. `DROP TABLE {full_name}_tmp` 

otherwise the failure of overwrite will leave the table in a corrupt state.